### PR TITLE
add error handler modules.

### PR DESCRIPTION
**Pull request by kekuso**
Experimenting with error handler modules to keep the code DRY.

### CODE REVIEW

File: prototypes/emptyParamErrorHandler.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+const emptyParamErrorHandler = (arg) => {
+    if (!arg[0]) {
+        throw Error('Undefined local variable');
+    } else {
```
Comment (mfuroyama) [inline prototypes/emptyParamErrorHandler.js]: Actually, you don't need the `else` clause here.

File: prototypes/paramErrorHandler.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+const paramErrorHandler = (arg) => {
+    if (arg[0] === null || undefined) {
```
Comment (mfuroyama) [inline prototypes/paramErrorHandler.js]: I think statement isn't doing what you intended it to do.  It seems like what you want/need is:
`if (arg[0] === null || arg[0] === undefined) {`

You also might want to consider using the Underscore/Lodash `_.isEmpty()` function.

File: prototypes/paramErrorHandler.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        throw Error('simple typed value must be NUMBER or STRING, not null');
+    } else if (arg[0] === [[]]) {
+        throw Error('Passed in array value of zero length');
+    } else {
```
Comment (mfuroyama) [inline prototypes/paramErrorHandler.js]: Same thing here.  You don't need this last `else` clause.

File: prototypes/paramErrorHandler.js (hunk 2 above)
Comment (EdisonTing) [inline prototypes/paramErrorHandler.js]: +1

File: prototypes/paramErrorHandler.js (hunk 3 above)
Comment (EdisonTing) [inline prototypes/paramErrorHandler.js]: +1

File: prototypes/paramErrorHandler.js (hunk 2 above)
Comment (kekuso) [inline prototypes/paramErrorHandler.js]: hmm but what if the function treats a null value and an empty value as two different things?
e.g. function(null) {...} returns "Error: null value detected"
but
function() {...} returns "Error: Undefined argument detected."
Should we just make one generic Error statement for null/undefined/empty arguments?

Comment (kekuso) [inline prototypes/paramErrorHandler.js]: Also, some functions accept a blank array argument, but not a null argument.

Comment (kekuso) [inline prototypes/paramErrorHandler.js]: isemptyscreenshot

Comment (mfuroyama) [inline prototypes/paramErrorHandler.js]: Ahh, I see. Yeah, if you need to check for different types of _falsy-ness_, then yeah, don't use `_.isEmpty()`, which is more useful for checking global validity. Disregard my comment about considering `_.isEmpty()`

The individual type checking is okay but definitely fix the part where you check for `undefined`

Comment (kekuso) [inline prototypes/paramErrorHandler.js]: isemptyscreenshot2

Comment (kekuso) [inline prototypes/paramErrorHandler.js]: I think I'll make one error module with isEmpty() check, and one more that only checks for null value.

**Comment by kekuso** [comment]
refactoring!

**Pull request closed without merging**